plan/planbuilder.go

```diff
@@ -361,37 +353,62 @@ func splitWhere(where ast.ExprNode) []ast.ExprNode {
 }
 
 func (b *planBuilder) buildShow(show *ast.ShowStmt) Plan {
-	var p Plan
-	p = &Show{
-		Tp:     show.Tp,
-		DBName: show.DBName,
-		Table:  show.Table,
-		Column: show.Column,
-		Flag:   show.Flag,
-		Full:   show.Full,
-		User:   show.User,
-	}
+	var resultPlan Plan
+	p := &Show{
+		Tp:              show.Tp,
+		DBName:          show.DBName,
+		Table:           show.Table,
+		Column:          show.Column,
+		Flag:            show.Flag,
+		Full:            show.Full,
+		User:            show.User,
+		baseLogicalPlan: newBaseLogicalPlan("Show", b.allocator),
+	}
+	resultPlan = p
+	p.initID()
+	p.self = p
 	switch show.Tp {
 	case ast.ShowProcedureStatus:
-		p.SetFields(buildShowProcedureFields())
+		p.SetSchema(buildShowProcedureSchema())
 	case ast.ShowTriggers:
-		p.SetFields(buildShowTriggerFields())
+		p.SetSchema(buildShowTriggerSchema())
 	default:
-		p.SetFields(show.GetResultFields())
+		p.SetSchema(expression.ResultFieldsToSchema(show.GetResultFields()))
 	}
-	var conditions []ast.ExprNode
+	for i, col := range p.schema {
+		col.Position = i
+	}
+	var conditions []expression.Expression
 	if show.Pattern != nil {
-		conditions = append(conditions, show.Pattern)
+		expr, _, _, err := b.rewrite(show.Pattern, p, nil, false)
+		if err != nil {
+			b.err = errors.Trace(err)
+			return nil
+		}
+		conditions = append(conditions, expr)
 	}
 	if show.Where != nil {
```

Comment:
we should pass the ast.where to buildSelection. But in this case, we use both ast.PattenLikeExpr and ast.Where to constuct selection.